Documentation: Document ingester read path limiting (#7110)
grafana/mimir: docs/sources/mimir/configure/configure-resource-utilization-based-ingester-read-path-limiting.md

---
aliases:
  - ../operators-guide/configure/configure-resource-utilization-based-ingester-read-path-limiting/
description: Learn how to configure Grafana Mimir for resource utilization based ingester read path limiting.
menuTitle: Resource utilization based ingester read path limiting
title: Configure resource utilization based ingester read path limiting
weight: 120
---

> # Configure resource utilization based ingester read path limiting

Reviewer: This doc page lacks some important information (e.g. metrics) which are documented in `docs/sources/mimir/references/architecture/components/ingester.md` instead. I think we should document the feature only in 1 place, either here or in `docs/sources/mimir/references/architecture/components/ingester.md`. Given it's not a critical feature (I mean something you absolutely need to be aware of) I would suggest to move all the doc here, and completely remove it from `docs/sources/mimir/references/architecture/components/ingester.md` (it's already listed among the list of production tips, which is good).

Author: Done.

> As an **experimental** feature, Grafana Mimir allows you to configure limits for CPU and/or memory utilization which can cause the ingester to reject incoming read requests in order to protect the write path. The ingester's model of memory utilization

Reviewer: [non blocking] Same comment about experimental here (see the other comment).

Author: Donie, thanks.

---
aliases:
  - ../operators-guide/configure/configure-resource-utilization-based-ingester-read-path-limiting/
description: Learn how to configure Grafana Mimir for resource utilization based ingester read path limiting.
menuTitle: Resource utilization based ingester read path limiting
title: Configure resource utilization based ingester read path limiting
weight: 120
---

# Configure resource utilization based ingester read path limiting

> As an **experimental** feature, Grafana Mimir allows you to configure limits for respectively CPU and/or memory utilization at
> which point the ingester should reject incoming read requests, to protect the write path. The ingester's model of memory utilization
> corresponds to the Go heap size, and it tracks (a sliding window average of) the process' CPU utilization.

Reviewer 2: A phrasing suggestion:

```suggestion
As an **experimental** feature, Grafana Mimir allows you to configure limits for CPU and/or memory utilization which can cause the ingester to reject incoming read requests in order to protect the write path. The ingester's model of memory utilization
corresponds to the Go heap size, which it tracks, along with a sliding window average of the process' CPU utilization.
```
